consensus/packets/interfaces.go

```diff
@@ -48,5 +50,31 @@ type ConsensusPacket interface {
 }
 
 func ExtractPacket(reader io.Reader) (ConsensusPacket, error) {
-	return nil, errors.New("not implemented")
+	header := PacketHeader{}
+	err := header.Deserialize(reader)
+	if err != nil {
+		return nil, errors.New("[ ExtractPacket ] Can't read packet header")
+	}
+
+	var packet ConsensusPacket
+	switch header.PacketT {
+	case Phase1:
+		packet1 := &Phase1Packet{}
+		err := packet1.DeserializeWithoutHeader(reader, &header)
```

Comment:
move `DeserializeWithoutHeader` to `Serializer` interface and remove copy-paste lines 63-66, 70-74

Reply:
It means that all structures will have to realize it. It's better make one more separate interface with only this function